### PR TITLE
#1401. If- and for-elements tests for maps and sets added

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Patterns/execution_pattern_if_case_element_A01_t08.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  Expect.setEquals({0, 2}, s2);
+
+  log = "";
+  var s3 = {
+    0,
+    if (Square(2, logger) case Square(area: unit4) when guard(false)) 1,
+    2
+  };
+  Expect.equals("Square.area:(4==4);guard(false);", log);
+  Expect.setEquals({0, 2}, s3);
```
Comment (eernstg) [inline LanguageFeatures/Patterns/execution_pattern_if_case_element_A01_t08.dart:69]: Is the `s3` test case an duplicate of the `s2` test case? I can't spot the difference.

File: LanguageFeatures/Patterns/execution_pattern_if_case_element_A03_t03.dart [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+///     c. Else, evaluate the else element if there is one and yield the result
+///       into the collection.
+///   ii. Else there is no guard clause. Evaluate the then element and yield the
+///     result into the collection.
+/// 4. Else the match failed. Evaluate the else element if there is one and
+///   yield the result into the collection.
+///
+/// @description Check that it is a compile-time error if the value yielded by
+/// pattern-if-case element is not assignable to the collection element type
+/// @author sgrekhov22@gmail.com. Test a set literal
```
Comment (eernstg) [inline LanguageFeatures/Patterns/execution_pattern_if_case_element_A03_t03.dart:22]: I can't see any set literals here?

**Review by eernstg — COMMENTED** [review]
I think LanguageFeatures/Patterns/execution_pattern_if_case_element_A03_t03.dart needs to be adjusted (looks like it is testing one thing, but the `@description` describes another thing, and the whole test is probably a duplicate of something which already exists).

A handful of other things: I think a particular outcome which is similar to the Fibonacci sequence needs a comment (this affects 5-6 libraries, I think). It's good, though, that we test the case where one key/value pair is _overwritten_ by another one in the same collection literal. We might very well not have covered that otherwise, so please don't change that. ;-)

File: LanguageFeatures/Patterns/execution_pattern_if_case_element_A03_t03.dart (hunk 2 above)
Comment (sgrekhov) [reply LanguageFeatures/Patterns/execution_pattern_if_case_element_A03_t03.dart:22]: Fixed. Thanx!

File: LanguageFeatures/Patterns/execution_pattern_if_case_element_A01_t08.dart (hunk 1 above)
Comment (sgrekhov) [reply LanguageFeatures/Patterns/execution_pattern_if_case_element_A01_t08.dart:69]: Indeed. Deleted a copy

**Review by eernstg — APPROVED** [review]
LGTM!
